### PR TITLE
Handle error gracefully on msgraph api photo load

### PR DESCRIPTION
**Pull request by msamad**
Signed-off-by: Mustansar Anwar ul Samad <mustansar.samad@gmail.com>

## Hey, I just made a Pull Request!

When using Microsoft Graph API to load users, the loading fails if any user's photo loading throws an error. This PR catches the error and logs a warning for the failed load.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by Fox32** [comment]
Hey @msamad, thanks for your contribution. The `MicrosoftGraphOrgReaderProcessor` was extracted into a separate package in a recent PR, you have to implement your changes there (https://github.com/backstage/backstage/tree/master/plugins/catalog-backend-module-msgraph).

**Comment by msamad** [comment]
Thanks @Fox32, I'll move my changes

**Comment by freben** [comment]
Hey for your information, unfortunately it seems there's now a conflict with some other change. Sorry about that. I'll get around to doing the full review later.

**Comment by msamad** [comment]
thanks @freben, I've resolved the conflict. The `CI / verify (14.x) (pull_request)` is failing though and when I try to do `yarn build:api-reports` locally, it fails with the following error, don't know how to resolve this.
```
## Processing packages\config
## Processing packages\config-loader
## Processing packages\core-app-api
## Processing packages\core-components
Error: "import * as ___ from ___;" is not supported yet for local files.
```

File: plugins/catalog-backend-module-msgraph/src/microsoftGraph/read.ts [hunk 1]
```diff
@@ -74,6 +75,7 @@ export async function defaultUserTransformer(
 }
 
 export async function readMicrosoftGraphUsers(
+  logger: Logger,
```
Comment (freben) [inline plugins/catalog-backend-module-msgraph/src/microsoftGraph/read.ts:78]: Can we move this into the `options` parameter instead? We have a bit of a new emerging standard of sorts, where we either try to make one single large options object for complex methods, or have a small number of very obviously "core" parameters that drive the functionality, and a smaller options object with all of the auxiliary things.

Bit of a nitpick, but we've seen functions/methods like this grow and grow over time and become really unwieldy and hard to call and evolve.

File: plugins/catalog-backend-module-msgraph/src/microsoftGraph/read.ts [hunk 2]
```diff
@@ -369,6 +376,7 @@ export function resolveRelations(
 }
 
 export async function readMicrosoftGraphOrg(
+  logger: Logger,
```
Comment (freben) [inline plugins/catalog-backend-module-msgraph/src/microsoftGraph/read.ts:379]: same here

**Review by freben — APPROVED** [review]

**Comment by freben** [comment]
helped fix the api reports and made the options change as well, waiting for build to turn green then 🚢

**Comment by msamad** [comment]
@freben oh wow, you fixed it for me, thank you very much. And sorry, I was hoping to fix it but couldn't manage it today. Thanks again for the great help, appreciate it. :)

**Comment by freben** [comment]
And we appreciate you getting this going! :)

**Pull request merged**